doc/elements-tx-format.md

```diff
@@ -0,0 +1,562 @@
+
+01 ................................... Flags (0x01 → witness data is present)
+|
+|                                      Input #1
+| 3d0bfb6e61989331d67dac3b6adf067a
+| fda20f504cabb1a7364853a68a0bb6df ... Outpoint TXID 
+| 00000040 ........................... Output Index (little-endian → 0x40000000 → second most significant bit is 1 → pegin)
+|
+| 00 ................................. ScriptSig length
+| | .................................. ScriptSig (empty: segwit transaction)
+|
+| ffffffff ........................... Sequence number: UINT32_MAX
+
+02 ................................... Num outputs
+|
+|                                      Output #1
+| 01 ................................. Asset header (0x01 → explicit, unblinded value)
+| 6d521c38ec1ea15734ae22b7c4606441
+| 2829c0d0579f0a713d1c04ede979026f ... Key used to generate asset ID
+|
+| 01 ................................. Amount header (0x01 → explicit, unblinded value) 
+| 00000000002b09c1 ................... Amount: 0.02820545 L-BTC
+|
+| 00 ................................. Nonce header (0x00 → null)
+|
+| 16 ................................. ScriptPubKey length (0x16 = 22)
+| | 00143921df9000812998ceff99b610
+| | 437734d552424f ................... ScriptPubKey
+|
+|                                      Output #2
+| 01 ................................. Asset header (0x01 → explicit, unblinded value)
+| 6d521c38ec1ea15734ae22b7c4606441
+| 2829c0d0579f0a713d1c04ede979026f ... Key used to generate asset ID
+|
+| 01 ................................. Amount header (0x01 → explicit, unblinded value) 
+| 0000000000000027 ................... Amount: 0.00000039 L-BTC
+|
+| 00 ................................. Nonce header (0x00 → null)
+|
+| 00 ................................. ScriptPubKey length
+| | .................................. ScriptSig (empty: segwit transaction)
+
+00000000 ............................. Locktime
+
+...................................... Input witnesses (1 per input)
+|
+|                                      Input witness #1
+| 00 ................................. Issuance amount range proof (null)
+| 00 ................................. Inflation keys range proof (null)
+| 02 ................................. Script witness stack length
+| | 47 ............................... Stack item #1 length (0x47 = 71 bytes)
+| | | 304402204f84bb59c2af17b76ba7
+| | | 1486a3fe4941829f0b5d23bda5c9
+| | | f49c449489693e5002206e1f3bba
+| | | 907e511e99bf317059264f89a56c
+| | | 50c56fefa82583b8e941b865facc
+| | | 01 ............................. Stack item #1
+| | 21 ............................... Stack item #2 length (0x21 = 33 bytes)
+| | | 02e776bdb5d8fc14d24b330c0efb
+| | | f34227cbba1cf25eb0778aa45f8b
+| | | 7cb3495046 ..................... Stack item #2
+| 06 ................................. Pegin witness stack length
+| | 08 ............................... Stack item #1 length  
+| | | e8092b0000000000 ............... Stack item #1
```

Comment:
@apoelstra Should be be describing the pegin structure here, amount, claim_script, genesis hash, peg-in transaction?  I know that Allen's tool does.

Reply:
Yeah, I think that we should. I'm tempted to say "it's a stack of byte vectors" but in this case every field has a clear meaning and fixed format, and it's simple enough to parse.